src/main/java/org/geysermc/generator/MappingsGenerator.java

```diff
@@ -615,6 +628,115 @@ public void generateParticles() {
+                } else if (state.getBlock() instanceof FlowerPotBlock) {
+                    alwaysConsume.add(blockStateToString(state)); // Contains checks for item, but will always consume the action
+                    continue;
+                } else if (state.getBlock() == Blocks.DRAGON_EGG) {
+                    alwaysConsume.add(blockStateToString(state)); // Teleports and will always consume the action
+                    continue;
+                } else if (state.getBlock() == Blocks.CAKE) {
+                    continue; // Depends on the player's hunger level
+                } else if (state.getBlock() instanceof SignBlock) {
+                    alwaysConsume.add(blockStateToString(state)); // Contains checks for item, but will always consume the action
+                    continue;
+                }
+
+                requiresAbilities.set(false);
+                abilities.mayBuild = true;
+
+                requiresItem.set(false);
+                item.set(ItemStack.EMPTY);
+
+                if (state.getBlock() instanceof BaseEntityBlock baseEntityBlock) {
+                    when(mockClientLevel.getBlockEntity(BlockPos.ZERO)).thenReturn(baseEntityBlock.newBlockEntity(BlockPos.ZERO, state));
+                } else {
+                    when(mockClientLevel.getBlockEntity(BlockPos.ZERO)).thenReturn(null);
+                }
+                when(mockClientLevel.getBlockState(new BlockPos(0, 0, 0))).thenReturn(state);
+
+                InteractionResult result = state.use(mockClientLevel, mockPlayer, InteractionHand.MAIN_HAND, blockHitResult);
+                if (!requiresItem.get()) {
+                    if (result.consumesAction() && requiresAbilities.get()) {
+                        abilities.mayBuild = false;
+                        InteractionResult result2 = state.use(mockClientLevel, mockPlayer, InteractionHand.MAIN_HAND, blockHitResult);
+                        if (result != result2) {
+                            requiresMayBuild.add(blockStateToString(state));
+                        }
+                    } else if (result.consumesAction()) {
+                        alwaysConsume.add(blockStateToString(state));
+                    }
+                }
+            } catch (Throwable e) {
+                // Ignore; this means the block has extended behavior we have to implement manually
+                // e.printStackTrace();
+            }
+        }
+
+        File mappings = new File("mappings/interactions.json");
+        if (!mappings.exists()) {
+            System.out.println("Could not find mappings submodule! Did you clone them?");
+            return;
+        }
+        try {
+            GsonBuilder builder = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping();
+            JsonWriter writer = new JsonWriter(new FileWriter(mappings));
+            writer.setIndent("\t"); // Tabs just to keep the diff nice for older mappings
```

Comment:
This comment here is unnecessary - it was for particles to keep the indents the same.